packages/node/src/methods.ts

```diff
@@ -23,76 +28,91 @@ export async function setupXkeysPanel(devicePathOrHIDDevice?: HID.Device | HID.H
 		  }
 		| undefined
 
-	if (!devicePathOrHIDDevice) {
-		// Device not provided, will then select any connected device:
-		const connectedXkeys = listAllConnectedPanels()
-		if (!connectedXkeys.length) {
-			throw new Error('Could not find any connected X-keys panels.')
-		}
-		// Just select the first one:
-		devicePath = connectedXkeys[0].path
-		device = new HID.HID(devicePath)
-
-		deviceInfo = {
-			product: connectedXkeys[0].product,
-			productId: connectedXkeys[0].productId,
-			interface: connectedXkeys[0].interface,
-		}
-	} else if (isHID_Device(devicePathOrHIDDevice)) {
-		// is HID.Device
+	try {
+		if (!devicePathOrHIDDevice) {
+			// Device not provided, will then select any connected device:
+			const connectedXkeys = listAllConnectedPanels()
+			if (!connectedXkeys.length) {
+				throw new Error('Could not find any connected X-keys panels.')
+			}
+			// Just select the first one:
+			devicePath = connectedXkeys[0].path
+			device = await HID.HIDAsync.open(devicePath)
+
+			deviceInfo = {
+				product: connectedXkeys[0].product,
+				productId: connectedXkeys[0].productId,
+				interface: connectedXkeys[0].interface,
+			}
+		} else if (isHID_Device(devicePathOrHIDDevice)) {
+			// is HID.Device
 
-		if (!devicePathOrHIDDevice.path) throw new Error('HID.Device path not set!')
+			if (!devicePathOrHIDDevice.path) throw new Error('HID.Device path not set!')
 
-		devicePath = devicePathOrHIDDevice.path
-		device = new HID.HID(devicePath)
+			devicePath = devicePathOrHIDDevice.path
+			device = await HID.HIDAsync.open(devicePath)
 
-		deviceInfo = {
-			product: devicePathOrHIDDevice.product,
-			productId: devicePathOrHIDDevice.productId,
-			interface: devicePathOrHIDDevice.interface,
+			deviceInfo = {
+				product: devicePathOrHIDDevice.product,
+				productId: devicePathOrHIDDevice.productId,
+				interface: devicePathOrHIDDevice.interface,
+			}
+		} else if (
+			typeof devicePathOrHIDDevice === 'object' &&
+			typeof (devicePathOrHIDDevice as any).devicePath === 'string'
+		) {
+			// (backwards compatibility): has { devicePath: string }
+
+			devicePath = (devicePathOrHIDDevice as any).devicePath
+			device = await HID.HIDAsync.open(devicePath)
```

Comment:
I added this blind implementation to keep some form of backwards compatibility in case someone provices that fabled HID.HID-with-a-devicePath that I don't remember where it can come from.

Reply:
when doing this, you will need to close the `HID.HID` handle before opening it in async, otherwise the async open will fail (on macos and linux) due to HID exclusivity rules.

Reply:
Ah ok, so it would probably be better to remove this case altogether then, since we actually don't know what `HID` device is actually is.